dashboard/src/components/providers/AppProvider.tsx

```diff
@@ -402,18 +394,27 @@ const middlewareReducer = async (
             backendUrl,
             alias,
             `
-            SELECT * FROM iasql_begin();
-          `,
-          );
-          await DbActions.run(
-            token,
-            backendUrl,
-            alias,
-            `
-            SELECT * FROM iasql_commit();
+            INSERT INTO aws_credentials (access_key_id, secret_access_key)
+            VALUES ('${awsAccessKeyId}', '${awsSecretAccessKey}');
           `,
           );
         }
+        await DbActions.run(
+          token,
+          backendUrl,
+          alias,
+          `
+          SELECT * FROM iasql_begin();
```

Comment:
shouldn't the begin be before?

Reply:
It actually doesn't matter on that front, just forcing `commit` to immediately trigger the loading of the AWS region data. The `aws_account` table's mapper isn't actually connected to anything on the cloud side. I think when I first wrote it I originally had the two select statements in the same request and someone must have separated them out. It was also put here because of the original conditional for before and after transactions (non-transactions was `iasql_apply`, with transactions was `iasql_begin .. iasql_commit`) and it reduced the number of conditionals needed from two to one this way.